### PR TITLE
Adding support for Unix time formatted dates in C#

### PR DESCRIPTION
**Pull request by tbombach**
Addressing issue #909
Adding a new known primary type for Unix time formatted dates (which is in seconds since Unix Epoch). Wiring up code namers to handle this primary type and set the correct.

Implementing the JsonConverter for C# and adding to templates for serializing/deserializing a DateTime object with the Unix time format. Adding path and body tests in C# to validate this functionality

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@tbombach__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (tbombach). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by jianghaolu** [comment]
Please pull from upstream master to get a hot fix in Java. Thanks!

File: AutoRest/AutoRest.Core/Utilities/Extensions.cs [hunk 1]
```diff
@@ -263,5 +263,54 @@ public static bool IsPrimaryType(this IType type, KnownPrimaryType typeToMatch)
             }
             return false;
         }
+
```
Comment (tbombach) [inline AutoRest/AutoRest.Core/Utilities/Extensions.cs]: @markcowl Mark, could you take a quick look at the changes I made in this last commit whenever it's convenient for you?

For context, MethodTemplateModel.GetSerializationSettingsReference() is getting too complex and failing a build rule for code complexity (and it's hard to read). I simplified this logic by adding methods to check if an IType either is a KnownPrimaryType or is a dictionary/sequence that is based on the KnownPrimaryType (previously these checks were repeated for each primary type that had special serialization behavior).

Tests are passing and the serialization is working the same as before. I'd like another set of eyes on it before I merge to make sure I kept the behavior the same with respect to the null checks. Thanks!

**Comment by amarzavery** [comment]
The changes look good to me.

**Pull request merged**